### PR TITLE
Refactor the Metrics API specification for instruments

### PR DESCRIPTION
**Pull request by jmacd**
## Changes

This consolidates several sections of text that were duplicated between the asynchronous and synchronous instrument specifications. This refactoring adds sections to the TOC where there were previously only named sections. This makes navigating easier and makes the structure of the general requirements clear to the reader.

There are NO SPECIFICATION CHANGES in this PR, only movement and less duplication. One approximately-duplicate paragraph was removed in favor of another pre-existing statement, both relating to duplicate observations from async callbacks.

Part of #2350.

### CODE REVIEW

File: specification/metrics/api.md [hunk 1]
```diff
@@ -280,16 +291,14 @@ instrument. It MUST be treated as an opaque string from the API and SDK.
...
-<a name="synchronous-instrument"></a>
+#### Synchronous/Asynchronous property
 
 * Synchronous instruments (e.g. [Counter](#counter)) are meant to be invoked
   inline with application/business processing logic. For example, an HTTP client
   could use a Counter to record the number of bytes it has received.
   [Measurements](#measurement) recorded by synchronous instruments can be
   associated with the [Context](../context/context.md).
 
-<a name="asynchronous-instrument"></a>
```
Comment (arminru) [inline specification/metrics/api.md:291]: At least those two anchors are referenced a couple of times in this document. You can change those occurrences to the new heading or leave the manual link next to them. This should also make the docfx check pass.

File: specification/metrics/api.md [hunk 2]
```diff
@@ -683,42 +711,7 @@ a strong reason not to do so. Please note that the name has nothing to do with
...
-* Return a list (or tuple, generator, enumerator, etc.) of `Measurement`s.
-* Use an observable result argument to allow individual `Measurement`s to be reported.
-
-User code is recommended not to provide more than one `Measurement` with the
-same `attributes` in a single callback. If it happens, the
-[SDK](./README.md#sdk) can decide how to handle it. For example, during the
-callback invocation if two measurements `value=3.38, attributes={cpu:1, core:2}`
-and `value=3.51, attributes={cpu:1, core:2}` are reported, the SDK can decide to
-simply let them pass through (so the downstream consumer can handle
-duplication), drop the entire data, pick the last one, or something else. The
```
Comment (arminru) [inline specification/metrics/api.md:713]: This paragraph went missing.

File: specification/metrics/api.md [hunk 3]
```diff
@@ -976,49 +962,13 @@ note that the name has nothing to do with [asynchronous
...
-* Use an observable result argument to allow individual `Measurement`s to be
-  reported.
-
-User code is recommended not to provide more than one `Measurement` with the
-same `attributes` in a single callback. If it happens, the
-[SDK](./README.md#sdk) MAY decide how to handle it. For example, during the
-callback invocation if two measurements `value=1, attributes={pid:4,
-bitness:64}` and `value=2, attributes={pid:4, bitness:64}` are reported, the SDK
-can decide to simply let them pass through (so the downstream consumer can
-handle duplication), drop the entire data, pick the last one, or something else.
```
Comment (arminru) [inline specification/metrics/api.md:1012]: This paragraph went missing.

File: specification/metrics/api.md (hunk 1 above)
Comment (arminru) [reply specification/metrics/api.md:291]: They're also referenced from `supplementary-guidelines.md`.

File: specification/metrics/api.md [hunk 4]
```diff
@@ -314,13 +346,36 @@ asynchronous instrumentation creation, it MUST return something (e.g.,
...
+
 Callback functions SHOULD NOT take an indefinite amount of time.
 
 Callback functions SHOULD NOT make duplicate observations from asynchronous
 instrument callbacks.  The resulting behavior when a callback observes
 multiple values for identical instrument and attributes is explicitly
 not specified.
 
+[OpenTelemetry API](../overview.md#api) authors MAY decide what is the idiomatic
+approach for capturing measurements from calblack functions. Here are some examples:
```
Comment (pirgeo) [inline specification/metrics/api.md:362]:
```suggestion
approach for capturing measurements from callback functions. Here are some examples:
```

File: specification/metrics/api.md (hunk 3 above)
Comment (jmacd) [reply specification/metrics/api.md:1012]: There is another passage of text that had been written in an earlier PR. 
Line 356-
> Callback functions SHOULD NOT make duplicate observations from asynchronous instrument callbacks.  The resulting behavior when a callback observes multiple values for identical instrument and attributes is explicitly not specified.

In #2361 this text is reformulated to apply to documentation for the user.

File: specification/metrics/api.md (hunk 2 above)
Comment (jmacd) [reply specification/metrics/api.md:713]: There is another passage of text that had been written in an earlier PR. 
Line 356-
> Callback functions SHOULD NOT make duplicate observations from asynchronous instrument callbacks.  The resulting behavior when a callback observes multiple values for identical instrument and attributes is explicitly not specified.

In #2361 this text is reformulated to apply to documentation for the user.

File: specification/metrics/api.md (hunk 1 above)
Comment (jmacd) [reply specification/metrics/api.md:291]: I updated the links to use the new #asynchronous-instrument-api and #synchronous-instrument-api sections.

**Review by MrAlias — APPROVED** [review]

**Pull request merged**